tests/integration/test_using_cached_dependencies.py

```diff
@@ -82,3 +98,7 @@ def test_using_cached_dependencies(test_env, tmpdir):
     first_deps = utils.make_list_of_packages_hashable(first_response.data["dependencies"])
     second_deps = utils.make_list_of_packages_hashable(second_response.data["dependencies"])
     assert first_deps == second_deps
+    # clean ups
+    if test_env["cached_dependencies"]["test_repo"].get("use_local"):
+        # this has been converted to a absolute path string
+        shutil.rmtree(test_env["cached_dependencies"]["test_repo"]["ssh_url"])
```

Comment:
If the test fails on some assertion this clean up won't happen. Maybe we can use try/finally block for this too?

Reply:
@ejegrova I moved the test into a test class and added proper setup and teardown methods.